core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2035,6 +2039,13 @@ protected Scope visitAliasedRelation(AliasedRelation relation, Optional<Scope> s
 
             RelationType descriptor = relationType.withAlias(relation.getAlias().getValue(), aliases);
 
+            boolean canProcessHiddenColumn = aliases == null;
+            Streams.forEachPair(
+                    descriptor.getAllFields().stream(),
+                    relationType.getAllFields().stream()
+                            .filter(field -> canProcessHiddenColumn || !field.isHidden()),
```

Comment:
How can we be sure that these two streams are in lock-step?

Reply:
Have added additional checks.